File: 자바스크립트/N으로표현.js
function solution(N, number) {
    const set = new Array(8).fill().map(() => new Set());
    // [set() for x in range(8)]
    for (let i = 0; i < 8; i++) {
        set[i].add(Number(N.toString().repeat(i + 1)));
        for (let j = 0; j < i; j++) {
            for (const arg1 of set[j]) {
                for (const arg2 of set[i - j - 1]) {
                    set[i].add(arg1 + arg2);
                    set[i].add(arg1 - arg2);
                    set[i].add(arg1 * arg2);
                    set[i].add(arg1 / arg2); // 자스는 0 나누는거 상관없음 -> infinity
                }
            }
        }
        if (set[i].has(number)) return i + 1;
    }
    return -1;
}

File: 파이썬/N으로표현.py
def solution(N, number):
    s = [set() for x in range(8)]
    for i, v in enumerate(s, 1):
        v.add(int(str(N)*i))
    for i in range(8):
        for j in range(i):
            for op1 in s[j]:
                for op2 in s[i-j-1]:
                    s[i].add(op1 * op2)
                    s[i].add(op1 - op2)
                    s[i].add(op1 + op2)
                    if op2 != 0:
                        s[i].add(op1 // op2)
        if number in s[i]:
            return i+1
    return -1
